forms/jqm/elements/date.js

```diff
@@ -83,7 +84,19 @@ define(function (require) {
     },
 
     onDateVChange: function (event) {
-      this.model.set('_date', $(event.target).val());
+      var dateFormat = this.model.mapDateFormats[this.model.attributes.dateFormat] || 'YYYY-MM-DD';
+      var value = $(event.target).val();
+
+      try {
+        this.model.isInvalidFormat(dateFormat, value);
+      } catch (err) {
+        window.console.log(err);
+        return;
+      }
+      if (value) {
+        value = moment(value, dateFormat).format('YYYY-MM-DD');
```

Comment:
sorry to be so nit picky but you should move the `var defaultFormat` and `var pickerFormat` declarations out of the functions and above the line

```
var DateElementView = ElementView.extend({
```

so they are only declared once, not every time the functions are run

and ron will probably ask that the names be uppercased like  `DEFAULT_FORMAT`